python-package/lightgbm/callback.py

```diff
@@ -16,7 +16,8 @@
 _EvalResultDict = Dict[str, Dict[str, List[Any]]]
 _EvalResultTuple = Union[
     List[_LGBM_BoosterEvalMethodResultType],
-    List[Tuple[str, str, float, bool, float]]
+    List[Tuple[str, str, float, bool, float]],
+    None
```

Comment:
Please don't include `None` in this type hint. That makes it difficult to use in other parts of the code where `None` isn't a valid value.